extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js

```diff
@@ -117,17 +114,12 @@ export class AmpAdXOriginIframeHandler {
       () => this.sendEmbedInfo_(this.baseInstance_.isInViewport())
     );
 
-    // TODO(bradfrizzell): Would be better to turn this on if
-    // A4A.isXhrEnabled() is false, or if we simply decide it is
-    // ok to turn this on for all traffic.
+    // Enable creative position observer if inabox experiment enabled OR
+    // adsense running on non-CDN cache where AMP creatives are xdomained.
     if (
-      getExperimentBranch(
-        this.win_,
-        ADSENSE_EXP_NAMES.UNCONDITIONED_CANONICAL
-      ) == ADSENSE_EXPERIMENTS.UNCONDITIONED_CANONICAL_EXP ||
-      getExperimentBranch(this.win_, ADSENSE_EXP_NAMES.CANONICAL) ==
-        ADSENSE_EXPERIMENTS.CANONICAL_EXP ||
-      isExperimentOn(this.win_, 'inabox-position-api')
+      isExperimentOn(this.win_, 'inabox-position-api') ||
+      (/^adsense$/i.test(this.element_.getAttribute('type')) &&
+        !isGoogleAdsA4AValidEnvironment(this.win_))
```

Comment:
do you know if AdSense using this position API? It's a bit weird to me that this opt-in based experiment is fully enabled to all AdSense A4A ineligible traffic?

Reply:
This was added to ensure that AMP creatives returned would have access to this information.  Since there is no way to know (or trust) that the creative is or is not AMP, we decided to just give access to all AdSense canonical creatives.

Reply:
thx for explaining. would you pls add it as a comment to the code?